pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -529,6 +538,22 @@ func isSCCAvailable(c kubernetes.Interface) (bool, error) {
 	return isResourceAvailable(c, "securitycontextconstraints", "security.openshift.io", "v1")
 }
 
+func isRunningKubernetesNMStateOperator(c kubernetes.Interface) (bool, error) {
```

Comment:
> Do we have to check if the NMState CR CNAO has is compatible with the installed CRD from the operator ?
> 

I don't think so. Versioning of the NMState API should take care of that. And the nmstate operator should handle that.

> Imaging that we are installing CNAO on openshift with old NMState operator we will apply a possible too new NMState CR and the opposite CNAO being old and openshift NMState operator newer.

So, an OpenShift cluster already has NMState operator installed.. CNAO would try to install a newer CR? That's an interesting scenario, but again, I think versioning would come into play there. I suppose we could detect in CNAO that the version we are trying to install is what we are trying to install... As for he CNAO being old and nmstate operator being newer... that should be handled by the nmstate operator.

> 
> Also in case the CR has being already applied I would do nothing to reduce possible problems.

If the CR is already applied, it should be a big no-op. But it would be interesting to find out what happens there.